swipl/src/dict.rs

```diff
@@ -206,8 +206,7 @@ unsafe impl<'a> TermPutable for DictBuilder<'a> {
 
         let tag_term = context.new_term_ref();
         let len = self.entries.len();
-        // TODO assert len is not too big
-        let value_terms = unsafe { fli::PL_new_term_refs(len as i32) };
+        let value_terms = unsafe { fli::PL_new_term_refs(len) };
```

Comment:
Probably we should have a `.try_into().unwrap()` here as well, right? This seems to be the way to make sure things work regardless of the generated signature of `PL_new_term_refs`.

Reply:
yep, that would make it work towards legacy swipl that expects i32